protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/DraggableSteps.tsx

```diff
@@ -0,0 +1,161 @@
+    () => ({
+      accept: DND_TYPES.STEP_ITEM,
+      canDrop: () => {
+        return true
+      },
+      drop: (item: DropType) => {
+        const draggedId = item.stepId
+        if (draggedId !== stepId) {
+          const overIndex = findStepIndex(stepId)
+          moveStep(draggedId, overIndex)
+        }
+      },
+      collect: (monitor: DropTargetOptions) => ({
+        handlerId: monitor.getHandlerId(),
+      }),
+    }),
+    [orderedStepIds]
+  )
+
+  drag(drop(ref))
+  return (
+    <Box
+      ref={ref}
+      style={{ opacity: isDragging ? 0.3 : 1 }}
+      data-handler-id={handlerId}
+    >
+      <ConnectedStepInfo stepNumber={stepNumber} stepId={stepId} />
+    </Box>
+  )
+}
+
+interface DraggableStepsProps {
+  orderedStepIds: StepIdType[]
+  reorderSteps: (steps: StepIdType[]) => void
+}
+export const DraggableSteps = (
+  props: DraggableStepsProps
+): JSX.Element | null => {
+  const { orderedStepIds, reorderSteps } = props
+  const { t } = useTranslation('shared')
+
+  const findStepIndex = (stepId: StepIdType): number =>
+    orderedStepIds.findIndex(id => stepId === id)
+
+  const moveStep = (stepId: StepIdType, targetIndex: number): void => {
+    const currentIndex = findStepIndex(stepId)
+
+    const currentRemoved = [
+      ...orderedStepIds.slice(0, currentIndex),
+      ...orderedStepIds.slice(currentIndex + 1, orderedStepIds.length),
+    ]
+    const currentReinserted = [
+      ...currentRemoved.slice(0, targetIndex),
+      stepId,
+      ...currentRemoved.slice(targetIndex, currentRemoved.length),
+    ]
+    if (confirm(t('confirm_reorder') as string)) {
+      reorderSteps(currentReinserted)
+    }
+  }
+
+  return (
+    <Flex
+      gridGap={SPACING.spacing4}
+      flexDirection={DIRECTION_COLUMN}
+      width="100%"
+    >
+      {orderedStepIds.map((stepId: StepIdType, index: number) => (
+        <DragDropStep
+          key={`${stepId}_${index}`}
+          stepNumber={index + 1}
+          stepId={stepId}
+          moveStep={moveStep}
+          findStepIndex={findStepIndex}
+          orderedStepIds={orderedStepIds}
+          // onStepContextMenu={makeStepOnContextMenu(stepId)}
+        />
+      ))}
+      <StepDragPreview />
+    </Flex>
+  )
+}
+
+const StepDragPreview = (): JSX.Element | null => {
+  const [{ isDragging, itemType, item, currentOffset }] = useDrag(() => ({
+    type: DND_TYPES.STEP_ITEM,
+    collect: (monitor: DragLayerMonitor) => ({
+      currentOffset: monitor.getSourceClientOffset(),
+      isDragging: monitor.isDragging(),
+      itemType: monitor.getItemType(),
+      item: monitor.getItem() as { stepId: StepIdType },
+    }),
+  }))
+
+  const savedStepForms = useSelector(stepFormSelectors.getSavedStepForms)
+  const savedForm = item && savedStepForms[item.stepId]
+  const { stepType, stepName } = savedForm || {}
+
+  if (
+    itemType !== DND_TYPES.STEP_ITEM ||
+    !isDragging ||
+    !stepType ||
+    !currentOffset
```

Comment:
For stepType, writing the condition explicitly would be good since stepType: StepType.

```suggestion
    stepType === undefined ||
    currentOffset === null
```